### DISCUSSION TITLE
Voice control cannot start play

### DISCUSSION THREAD

**Discussion by mcs1998** [music-assistant/support#2977] [Q&A] [answered]
Dear all,
I am trying for quite some time to get voice control to play music on my devices. The music plays ok when I use music assistant, but I get unexpected errors when using the home assistant conversation agent and with openai conversation agent the conversation essentially responds with "cannot play on this media player right now" or similar. I have made a dedicated ma specific conversation agent.

What is interesting is when started with music assistant, I can "stop" and "continue", but not play "next track" for example.
I enclose below the error message that may be most helpful, that is from the home assistant conversation agent.

Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/helpers/intent.py", line 142, in async_handle
    result = await handler.async_handle(intent)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/homeassistant/.homeassistant/custom_components/mass/intent.py", line 71, in async_handle
    mass_player = await self._async_get_matched_mass_player(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/homeassistant/.homeassistant/custom_components/mass/intent.py", line 140, in _async_get_matched_mass_player
    mass_player = MassPlayer(
                  ^^^^^^^^^^^
  File "/home/homeassistant/.homeassistant/custom_components/mass/media_player.py", line 228, in __init__
    super().__init__(mass, player_id)
  File "/home/homeassistant/.homeassistant/custom_components/mass/entity.py", line 29, in __init__
    provider = self.mass.get_provider(player.provider)
                                      ^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'provider'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/components/conversation/default_agent.py", line 367, in async_process
    intent_response = await intent.async_handle(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/homeassistant/lib/python3.12/site-packages/homeassistant/helpers/intent.py", line 150, in async_handle
    raise IntentUnexpectedError(f"Error handling {intent_type}") from err
homeassistant.helpers.intent.IntentUnexpectedError: Error handling MassPlayMediaOnMediaPlayer

Any help and pointers would b every much appreciated.
Cheers
mcs

**Comment by OzGav** [comment]
What happens when you type the command in? I am having no problems here but I use HA Cloud and the HA conversation agent

**Reply by mcs1998** [reply]
Thank you so much for the reply. I always type the commands, removes a variable that could go wrong. The error message above is with HA cloud and HA conversation agent. Response to the prompt is "An unexpected error occurred"
Thank you again
mcs

**Reply by OzGav** [reply]
Have a look at the docs which has some troubleshooting steps particularly with increasing the log level. Then there should be more useful info shown. You can also look at the voice assistant debug trace.

**Reply by mcs1998** [reply]
Thank you for the reply. Much appreciated. The traceback in the first post is with log level debug as suggested in the documentation. I added the relevant section to configuration.yaml. I also tried the suggestions in troubleshooting like adding an alias and making sure the areas are not called the same as the devices . I also tried to use players located in areas that only have  a single player.
The conversation debug is below, I do not see anything giving me a pointer but that probably does not mean a lot... :)

stage: done
run:
  pipeline: 01hajdtzqgj8dc869s0e6c6vk2
  language: en
  runner_data:
    stt_binary_handler_id: null
    timeout: 300
events:
  - type: run-start
    data:
      pipeline: 01hajdtzqgj8dc869s0e6c6vk2
      language: en
      runner_data:
        stt_binary_handler_id: null
        timeout: 300
    timestamp: "2024-10-02T06:56:35.812291+00:00"
  - type: intent-start
    data:
      engine: conversation.openai_conversation
      language: "*"
      intent_input: play that's amore on nest
      conversation_id: null
      device_id: null
    timestamp: "2024-10-02T06:56:35.812539+00:00"
  - type: intent-end
    data:
      intent_output:
        response:
          speech:
            plain:
              speech: >-
                It seems there was an issue playing "That's Amore" on the
                kitchen display. Would you like to try a different media player
                or check if the device is available?
              extra_data: null
          card: {}
          language: "*"
          response_type: action_done
          data:
            targets: []
            success: []
            failed: []
        conversation_id: 01J9608C40XWZQV2JJKH7Y3979
    timestamp: "2024-10-02T06:56:41.002674+00:00"
  - type: run-end
    data: null
    timestamp: "2024-10-02T06:56:41.003025+00:00"
intent:
  engine: conversation.openai_conversation
  language: "*"
  intent_input: play that's amore on nest
  conversation_id: null
  device_id: null
  done: true
  intent_output:
    response:
      speech:
        plain:
          speech: >-
            It seems there was an issue playing "That's Amore" on the kitchen
            display. Would you like to try a different media player or check if
            the device is available?
          extra_data: null
      card: {}
      language: "*"
      response_type: action_done
      data:
        targets: []
        success: []
        failed: []
    conversation_id: 01J9608C40XWZQV2JJKH7Y3979

**Reply by OzGav** [reply]
Attaching (don’t paste) more of the log will be helpful to see all parts of the intent matching. However, we can see from the above that the intent matching is working. The song title was correctly identified and I assume “nest” is an alias for the kitchen display?

**Reply by OzGav** [reply]
Also that conversation debug is that with ha cloud and the ha conversation agent?

**Reply by mcs1998** [reply]
yes, i gave it an alias nest. This is with ha cloud and openai, but I can do ha conversation agent as well. is that preferred?
Will put logs in files.

**Reply by OzGav** [reply]
Just if you do it with the HA conversation agent then I can recreate everything as that is what I use

**Reply by mcs1998** [reply]
I enclose the log with HA conversation.
[log_conversation.txt](https://github.com/user-attachments/files/17225074/log_conversation.txt)
Thank you so much for your help

**Reply by OzGav** [reply]
Thanks I’ll have a look at that in a couple of hours. Just as some more fault finding what if you have music playing on the device and you try “pause the nest” or “set the nest volume to 50”

**Reply by mcs1998** [reply]
pause the nest and setting the volume works when I start playing with musicassistant

**Reply by OzGav** [reply]
OK great. It might take me a couple of days to get back to you as I am busy with work.

**Reply by mcs1998** [reply]
Thank you very much for looking at this. Very much appreciated.

**Reply by OzGav** [reply]
OK I just tried "Play Thats amore on the snapweb client" and it worked fine. Can you restart HA then send the command "Play thats amore on the kitchen display" and attach all of the lines that relate to the request. You should see something like:
```
2024-10-05 00:48:40.159 DEBUG (MainThread) [homeassistant.components.conversation.agent_manager] Processing in en: Play thats amore on the snapweb client
2024-10-05 00:48:40.160 DEBUG (MainThread) [homeassistant.components.conversation.default_agent] Rebuilt trigger intents: {'language': 'en', 'intents': {'0': {'data': [{'sentences': ['Give me {query} on [the] {name}']}]}}}
2024-10-05 00:48:40.166 DEBUG (MainThread) [homeassistant.components.conversation.default_agent] Exposed entities: [('Garage Door Open', 'Garage Door Open', {'domain': 'binary_sensor', 'device_class': 'garage_door'}), ('pantry_lights_toggle', 'pantry_lights_toggle', {'domain': 'script'}), [snip]
2024-10-05 00:48:40.169 DEBUG (MainThread) [homeassistant.components.conversation.default_agent] Created slot lists in 0.01 seconds
2024-10-05 00:48:40.361 DEBUG (MainThread) [homeassistant.components.conversation.default_agent] Recognize done in 0.19 seconds
2024-10-05 00:48:40.364 DEBUG (MainThread) [homeassistant.components.conversation] Processing: <thats amore >
2024-10-05 00:48:40.364 DEBUG (MainThread) [homeassistant.components.conversation.agent_manager] Processing in en: thats amore
```

**Reply by mcs1998** [reply]
Sorry, I was sleeping I am in Australia, so we probably have  a different timezone...
I have restarted and done "play that's amore on the kitchen display" and get a different error (no valid targets) than when I say " play that's amore on display kitchen" I enclose both logs from the debugger. Where do I get this text based logs you show at the top? The error logs just show the python logs?

Thank you again for looking at this
[on_kitchen_display_1.log](https://github.com/user-attachments/files/17263732/on_kitchen_display_1.log)
[on_display_kitchen.log](https://github.com/user-attachments/files/17263735/on_display_kitchen.log)

**Reply by OzGav** [reply]
What is the actual name of the device? I did a line by line compare of the log to my working install and nothing jumped out. Do you have anything non standard with your network or your HA or MA install.

**Reply by mcs1998** [reply]
Hi,
The device is called Display kitchen and the device id's are media_player.display_kitchen and media_player.ma_display_kitchen in music assistant ( to be able to use the intent scripts as described in https://github.com/orgs/music-assistant/discussions/2176).
Homeassistant is running on a debian machine running as supervised. Music assistant is running in  a docker container on another debian machine. They computers are PC engines APU2. I also have three other speakers ( one display and two Yamaha soundbars ) where the behaviour is the same.
Thank you again for the help

**Reply by OzGav** [reply]
Can you restart HA then send the command "Play thats amore on the display kitchen" and attach all of the HA log lines that relate to the request. (See above for my example of the lines to include) And just confirming you have exposed these media players to assist.

**Reply by mcs1998** [reply]
Restarted and log attached. Yes I have exposed the players to assist.
Thank you very much
[
[hass_20241007.log](https://github.com/user-attachments/files/17272784/hass_20241007.log)
](url)

**Comment by mcs1998** [comment]
I must have missed a line in the above log.
I repeated the "Play thats amore on the display kitchen" and enclose the result
[hass_20241008.log](https://github.com/user-attachments/files/17290935/hass_20241008.log)

**Reply by OzGav** [reply]
I am wondering if your AI Integration is still interfering as if you look at my example vs your I see a number of additional lines from `homeassistant.components.conversation.default_agent` but lets just break this down further to check each component. 

You can create a voice assistant to just return the JSON to make sure that is working correctly. So please create another voice assistant and put in the conversation agent field the one you have added to the MA Integration. It should look something like this
![image](https://github.com/user-attachments/assets/217998b3-8261-4ad6-b643-00e5ca7b0cbf)

Then when you use Assist change to this voice assistant and you should see something like this:

![image](https://github.com/user-attachments/assets/8eca6e9e-abe3-4706-a4ee-899864dc2b6b)

**Comment by mcs1998** [comment]
Hi,

This is what I get when I do what you suggest above:
![Screenshot from 2024-10-10 17-36-28](https://github.com/user-attachments/assets/a4941c52-0591-45dc-a47e-7c363c2db11c)
Looks ok to me and is similar to what you show

Thanks

**Reply by OzGav** [reply]
@jozefKruszynski Any thoughts?

**Reply by OzGav** [reply]
@mcs1998 What music providers do you have configured? And tell me again exactly what this player is that you are using and how is it configured in MA? And please confirm you are running the latest version of HA.

**Reply by mcs1998** [reply]
I have the "filesystem" music provider installed, but also tried PLEX. Same effect.
![hass_2](https://github.com/user-attachments/assets/acf61239-ced6-4a35-8c18-75d7a0b2aa0d)

I am currently running 2024.10.2
Thank you for all your help and not giving up.

**Reply by OzGav** [reply]
And what player providers are you using? How many different players have you tried?

**Reply by mcs1998** [reply]
I have only used the Homeassistant player provider, with 2 google cast integrations in HA, and 2 yamaha soundbars. All the same issues.

**Reply by OzGav** [reply]
OK then it might be a HA player provider problem. Please add the cast devices natively to MA and test those. 

edit: You should be doing that anyway. Using the HA player provider is a fallback option

**Reply by mcs1998** [reply]
I tried chromecast player provider and UPNP just now, has unfortunately the same effect. Errors are the same.

**Reply by OzGav** [reply]
Can you just post the error again when using the MA Cast player just so I can see we are talking about the same thing. Also so we are both on the same sheet can you provide the assist command you typed and then show a screen capture of the HA SETTINGS>>ENTITIES which shows the name of the player you used. Finally provide a screenshot of the MA UI which shows the track you are trying to play.

**Reply by mcs1998** [reply]
![Screenshot from 2024-10-17 16-59-58_1](https://github.com/user-attachments/assets/4f5869a9-ac25-4b68-8cee-18096def9cce)
![Screenshot from 2024-10-17 17-24-57_1](https://github.com/user-attachments/assets/046d922e-d645-490b-ad6a-cb1de9ed5989)
![Screenshot from 2024-10-17 16-56-03_1](https://github.com/user-attachments/assets/0b372ef4-87a1-4fdd-8367-4662b40bf05b)
[hass_20251017_1.log](https://github.com/user-attachments/files/17407330/hass_20251017_1.log)
Hope this helps, please let me know if you need anything else

**Reply by OzGav** [reply]
You have two players with identical friendly names. If that isn't the problem it certainly isn't going to help.

edit: But I am not convinced that is it based on the error. I still wonder if your AI integrations are upsetting something. I understand you may not want to do this but if you could disable all of those except for the MA specific one that would remove some more variables

**Reply by mcs1998** [reply]
Hi,

I have deleted the other assistants and just left the one that I created as above. Same response as before:
![image](https://github.com/user-attachments/assets/977a007c-dd89-4dc2-bf1a-75ffcb3b6111)
Or did you mean delete all the assistants in HA?
Cheers

**Reply by OzGav** [reply]
No I meant the AI Integrations. All you should have left is one Open AI Conversation integration for use by MA. 

edit: I am just trying to get you in the same config as me

**Comment by mcs1998** [comment]
I am happy to do that, but do not know how. Delete is greyed out for the default assistant Any pointers would be appreciated

**Reply by mcs1998** [reply]
Have used this integration: 
https://github.com/tronikos/default_config_disabler
and disabled conversation. After a restart assist is still there but no conversations.
Then use my conversation manager I created for MA and get:
![Screenshot from 2024-10-18 19-49-51_1](https://github.com/user-attachments/assets/b3382f02-1738-414b-9062-fb49be2d7fbc)

**Reply by OzGav** [reply]
OK sorry maybe I misunderstood. Are you using a LLM integration or are you just using the Open AI Conversation Integration?

**Reply by mcs1998** [reply]
I have originall used Open Ai conversation but after your advice used the homeassistant conversation. Screenshot below
![Screenshot from 2024-10-19 07-43-14_1](https://github.com/user-attachments/assets/d265191a-3835-4e4a-9be5-69204e02b82a)
Sorry to be such a pain

**Reply by OzGav** [reply]
OK the above looks good. Now your probably had this before (or still do) but setup the Open AI Conversation integration with the our prompt and these settings 
![image](https://github.com/user-attachments/assets/bebfa605-aeb2-4707-aca5-fea959ae98f9)

Then make sure the MA Integration is configured to use that Integration
![image](https://github.com/user-attachments/assets/942a6bd6-679b-4376-90ff-b057c1c86678)

**Comment by mcs1998** [comment]
Done that. Only difference was in "Control Home Assistant" I had ticked "assist". Changed that but I get the same error.

**Reply by OzGav** [reply]
I am running out of ideas. Can you try "Play That's Amore by Dean Martin on the Dining Room Display"

If that fails can you install the Snapcast provider. Open a browser and type in `<YOUR_MA_IP>:1780` Press the triangle to turn on the player. You should find in MA there is now a player called `Snapweb Client` . Check that it has appeared in HA. Now try "Play That's Amore by Dean Martin on Snapweb Client"

**Reply by mcs1998** [reply]
Unfortunately, same error for both cases. I also updated MA integration to the latest this morning. 
Thanks for all the help

**Reply by OzGav** [reply]
Something weird is happening with the connection to your players then. I’ll have to think about it some more.

**Comment by mcs1998** [comment]
Thank you so muchOn Oct 21, 2024 08:20, OzGav ***@***.***> wrote:
Something weird is happening with the connection to your players then. I’ll have to think about it some more.

**Reply by mcs1998** [reply]
Now that is interesting. I tried "play muriel by tom waits on dining room display"  and it works????? "play That's amore by dean martin on dining room display" does not work? muriel by tom waits on dining room display
![image](https://github.com/user-attachments/assets/91f98e3d-8fe8-442e-befc-8e115f1264d1)


I also have tried with the music assistant and the Json is pretty much identical.
![image](https://github.com/user-attachments/assets/e4fffee9-582b-4af6-8119-fc431dc494aa)

Strange!
Thank you so much for all the help.

**Reply by OzGav** [reply]
Maybe for some reason MA is not finding that track. Can you keep testing and try more tracks for that dean Martin album. Try playing an artist. Try playing an album. Try lots of things and see if you can find a pattern

**Reply by OzGav** [reply]
OK a complete rewrite of the voice components has been done. Please go back and review the documentation for the new requuirements and see if this is resolved.

**Reply by OzGav** [reply]
@mcs1998 Changes will happen again when the integration moves into HA core so if this still isn't working wait until then

**Comment by mcs1998** [comment, marked as answer]
Hi all,
Thanks to Ozgav for all the changes implemented, really appreciated. I was struggling recently setting up the new versions of the integration, the integration continuously initialized and did not find any players. Took me some time to figure out I had to update the music-assistant server as well. Now with the new version of musicassistant integration (2024.12.1) from HACS assist works and I can play music with assist. Success!!!!! The same "an unexpected error occurred" appears with the HA core integration, so for now I will resist the change from HACS to core despite  the warnings.
Thanks again for all the updates and hard work.
Cheers
mcs1998
